nodebuilder/settings.go

```diff
@@ -33,6 +34,31 @@ func WithBootstrappers(peers p2p.Bootstrappers) fx.Option {
 	return fx.Replace(peers)
 }
 
+// WithPyroscope enables pyroscope profiling for the node.
+func WithPyroscope(endpoint string, nodeType node.Type) fx.Option {
+	return fx.Options(
+		fx.Invoke(func(peerID peer.ID) error {
+			_, err := pyroscope.Start(pyroscope.Config{
+				ApplicationName: "celestia.da-node",
+				ServerAddress:   endpoint,
+				Tags: map[string]string{
+					"type":   nodeType.String(),
+					"peerId": peerID.String(),
+				},
+				Logger: nil,
```

Comment:
Should we pass the logger?

Reply:
No, because its annoying, we can leave it as nil